xbmc/dbwrappers/dataset.h

```diff
@@ -49,49 +49,57 @@ constexpr unsigned int DB_CONNECTION_DATABASE_NOT_FOUND = 2;
 class Database
 {
 protected:
-  bool active;
-  bool compression;
-  std::string error, // Error description
-      host, port, db, login, passwd, //Login info
-      sequence_table, //Sequence table for nextid
-      default_charset, //Default character set
-      key, cert, ca, capath, ciphers; //SSL - Encryption info
+  bool active{false};
+  bool compression{false};
+  std::string error; // Error description
+  std::string host;
+  std::string port;
+  std::string db;
+  std::string login;
+  std::string passwd; // Login info
+  std::string sequence_table{"db_sequence"}; // Sequence table for nextid
+  std::string default_charset; // Default character set
+  std::string key;
+  std::string cert;
+  std::string ca;
+  std::string capath;
+  std::string ciphers; // SSL - Encryption info
   unsigned int connect_timeout; // seconds
 
 public:
   /* constructor */
   Database();
   /* destructor */
   virtual ~Database();
-  virtual Dataset* CreateDataset() const = 0;
+  virtual Dataset* CreateDataset() = 0;
```

Comment:
const removed?

Reply:
yes, intentionally, look at implementations, which needed a `const_cast` due to this.

https://github.com/xbmc/xbmc/blob/master/xbmc/dbwrappers/mysqldataset.cpp#L64